### PR TITLE
[types][dist] Bump deps, add types

### PR DESCRIPTION
**Pull request by decompil3d**
This PR adds typings and also bumps deps. The two changes are severable, so if you'd prefer to keep support for very old unsupported node.js versions, let me know and I can move back to the old deps.

### CODE REVIEW

**Comment by terinjokes** [comment]
LGTM.

File: .travis.yml [hunk 1]
```diff
@@ -1,10 +1,8 @@
 sudo: false
 language: node_js
 node_js:
-  - "5"
-  - "4"
-  - "iojs"
-  - "0.12"
-  - "0.10"
+  - "10"
```
Comment (terinjokes) [inline .travis.yml:5]: It looks like the union of sinon's and assume's tested Node.js versions includes Node.js 8. It might make sense to test that here.

**Review by jcrugzz — APPROVED** [review]

**Review by terinjokes — APPROVED** [review]

**Comment by terinjokes** [comment]
I think Travis CI is broken.

**Comment by decompil3d** [comment]
@terinjokes Jarrett tells me that you might want to hand off this package. If you want, you can transfer it over to me for now -- my username is the same in npm as well. Then I can handle the publish of this latest change.

**Comment by terinjokes** [comment]
I've added you here and there. Once you have a final place for this, I'm happy to transfer the repository.

**Comment by decompil3d** [comment]
Cool -- will go ahead and do the version bump and publish then.

**Comment by decompil3d** [comment]
@terinjokes Looks like somehow the npm package shows in my user's list but I don't have publish perms on it. I've pushed the updated version tag here, just need publish perms so I can get it updated on npm.

**Comment by terinjokes** [comment]
There might be a delay in updating permissions in the registry, I don't know the current architecture anymore. Try again in a little bit, and if still doesn't work I can raise a ticket.

**Pull request merged**